### ISSUE TITLE
Spread the word

### ISSUE THREAD

**Issue by steelbrain** [steelbrain/declarations#4]
`declarations` is a base package that allows package authors to use a super-easy API to show declarations to users.

API docs can be found in [the wiki](https://github.com/steelbrain/declarations/wiki/Provider-API)

**Comment by steelbrain** [comment]
Going to ping some people here

@lierdakil from `ide-haskell`
@devoncarew from `dartlang`
@david-driscoll from `omnisharp-atom`
@basarat from `atom-typescript`
@AsaAyers from `hyperclick-js`
@rameshvarun from `love-ide`
@Krzysztof-Cieslak from `ionide-fsharp`
@raymundl from `file-hyperclick`

Please ping anyone else you think would be interested in this package

**Comment by lierdakil** [comment]
Okay, so I read through wiki, readme and some code and I still don't understand the purpose of this package. Maybe I'm just having one of those dumb days, but... A screenshot or something could be nice?

**Comment by lierdakil** [comment]
Just a thought, if you're trying to implement generalized 'go-to-declaration' package, going _back_ could be a good addition (not sure if you've done this already though).

**Comment by lierdakil** [comment]
Also a common theme with 'go-to-declaration' is being able to show list of declarations in current file and/or in whole project. Could be useful too. Not sure if it's in scope for this package though.

**Comment by steelbrain** [comment]
@lierdakil Yes, this is a generalized `go-to-declaration` package, the `visibleRange` in `getDeclarations` argument is for providers that do regex matching, so they can provide that range to `textEditor.scanTextInBufferRange` and don't have to scan the entire editor

**Comment by Krzysztof-Cieslak** [comment]
I definitely like idea of having such packages adding more IDE-like common features. 

What I would like to see in this package (examples from VSCode as they have those features as part of API):
1. Go-to-declaration for current symbol (https://code.visualstudio.com/docs/editor/editingevolved#_go-to-definition)
2. Show all occurrences of current symbol 
3. Peek definition for current symbol (https://code.visualstudio.com/docs/editor/editingevolved#_peek)
4. Show all symbols in file (https://code.visualstudio.com/docs/editor/editingevolved#_goto-symbol) 
5. Show all symbols in project (https://code.visualstudio.com/docs/editor/editingevolved#_open-symbol-by-name)

**Comment by steelbrain** [comment]
Currently this package underlines all known declarations when the user holds ctrl, clicking any of them takes the user to the declaration. Another way of jumping is to trigger the intentions list (alt + enter) and select jump to declaration from the list

**Comment by Krzysztof-Cieslak** [comment]
Holding ctrl sounds reasonable. 

Not sure if I like intentions list at all, will need to play with it bit more (Another question - how hard is dependency on intentions package? Will this package work without intentions one?). 

I would also like to see something using [Select List View](https://github.com/atom/atom-space-pen-views#selectlistview) - it will provide same UX as in Code.

**Comment by steelbrain** [comment]
The intentions dependency is pretty hard, it handles both the highlights and the menu, this package listens on the declaration API and registers itself as a provider for both types of APIs in intentions and handles the low level work.

The current list implementation uses the same markup as the view you showed, it uses jsx instead of space-pen views though, but that's a deep implementation thing and shouldn't affect provider APIs
